Makefile

```diff
@@ -1,113 +1,99 @@
 
 # XC patch
 XC7PATCH=$(SYMBIFLOW_ARCH_DEFS)/build/third_party/prjxray/tools/xc7patch
 
+################ Symbiflow files dependencies ################
+
+BASE_DIR=build/xc7/archs/artix7/devices
+
+#TARGET?=xc7a50t-arty-swbut
+TARGET?=xc7a50t-basys3
+
+DEVICE=$(TARGET)-test
+
+# Is this target outputting boths?
+ARCH_TARGET?=file_xc7_archs_artix7_devices_$(TARGET)-roi-virt_synth_tiles_pinmap.csv
+ARCH_TIMING=$(SYMBIFLOW_ARCH_DEFS)/$(BASE_DIR)/$(TARGET)-roi-virt/arch.timing.xml
+ARCH_PINMAP=$(SYMBIFLOW_ARCH_DEFS)/$(BASE_DIR)/$(TARGET)-roi-virt/synth_tiles_pinmap.csv
+
+# RRGRAPH
+RR_GRAPH_TARGET?=artix7_$(TARGET)_test_rrxml_real
+RR_GRAPH?=$(SYMBIFLOW_ARCH_DEFS)/$(BASE_DIR)/rr_graph_$(TARGET)_test.rr_graph.real.xml
+
+# Convert FASM to frames
+FASM2FRAMES=PYTHONPATH=$(SYMBIFLOW_ARCH_DEFS)/third_party/prjxray:$(SYMBIFLOW_ARCH_DEFS)/third_party/prjxray/third_party/fasm python3 $(SYMBIFLOW_ARCH_DEFS)/third_party/prjxray/utils/fasm2frames.py
+
+# Place and route parameters, some are just added to speedup the computation
+VPR_PARAMS=--min_route_chan_width_hint 100 --max_criticality 0.0 --max_router_iterations 500 --routing_failure_predictor off --router_high_fanout_threshold -1 --constant_net_method route --route_chan_width 500 --clock_modeling route --place_algorithm bounding_box --enable_timing_computations off --allow_unrelated_clustering on --clustering_pin_feasibility_filter off --disable_check_route on --strict_checks off --allow_dangling_combinational_nodes on --disable_errors check_unbuffered_edges:check_route
+
+GENFASM_PARAMS=--route_chan_width 500
+
+################ Current project files and target ################
+
+# Maps the FPGA IOs to the top module verilog symbols.
+PCF=arty7.pcf
+
 # Flashing. Just use good 'ol openocd. Everyone should've installed that on their machine.
-OPENOCD_BIN?=$(CONDA_DIR)/bin/openocd
-DIGILENT_CONFIG=openocd-digilent-basys3.cfg
+DIGILENT_CONFIG=openocd-digilent-arty7.cfg
 
+PROJECT=blink
 
 all: top.bit
 
 %.eblif : %.v synth.tcl
-	symbiflow_arch_defs_SOURCE_DIR=$(SYMBIFLOW_ARCH_DEFS) \
-        OUT_EBLIF=$@ \
-        $(YOSYS_BIN) -q -p "tcl synth.tcl" -l $@.log $<
+	symbiflow_arch_defs_SOURCE_DIR=$(SYMBIFLOW_ARCH_DEFS) OUT_EBLIF=$@ \
+		$(YOSYS_BIN) -q -p "tcl synth.tcl" -l $@.log $<
 
-%_io.place : %.eblif
-	$(IOPLACE) --pcf $(PCF) --map $(MAP) --blif $^ --out $@
+%_io.place : %.eblif %.net
+	$(IOPLACE) --pcf $(PCF) --map $(ARCH_PINMAP) --blif $< --net $*.net  --out $@
 
 %.net: %.eblif
-	$(VPR_BIN) $(ARCH_TIMING_XML) $< --device $(DEVICE) --read_rr_graph $(RR_GRAPH) $(VPR_PARAMS) --pack > $*-pack.log
+	$(VPR_BIN) $(ARCH_TIMING) $< --device $(DEVICE) --read_rr_graph $(RR_GRAPH) $(VPR_PARAMS) --pack > $*-pack.log
 
 %.place: %.eblif %_io.place %.net
-	$(VPR_BIN) $(ARCH_TIMING_XML) $*.eblif --device $(DEVICE) --read_rr_graph $(RR_GRAPH) $(VPR_PARAMS) --fix_pins $*_io.place --place > $*-place.log
+	$(VPR_BIN) $(ARCH_TIMING) $*.eblif\
+		--device $(DEVICE) \
```

Comment:
I like the new organization with backslashs of these parameters here.

Reply:
(solved)